src/types/sync.rs

```diff
@@ -0,0 +1,26 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::ops::Range;
+
+use serde::{
+    Deserialize,
+    Serialize,
+};
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SyncRecord {
+    pub milestone_index: u32,
+    pub logged: bool,
+    pub synced: bool,
+}
+
+#[derive(Debug, Clone, Default, Serialize)]
+pub struct SyncData {
+    /// The completed(synced and logged) milestones data
+    pub completed: Vec<Range<u32>>,
+    /// Synced milestones data but unlogged
+    pub synced_but_unlogged: Vec<Range<u32>>,
+    /// Gaps/missings milestones data
+    pub gaps: Vec<Range<u32>>,
+}
```

Comment:
I guess these two structs keep track of the amount of data that is available on the permanode? I would remove them from the PR and add an issue to the repository to remember that functionality?

Reply:
They are fundamental to the design, as we cannot know when to request milestones unless we know which ones we have already gathered every message for.